task1/main.c

```diff
@@ -0,0 +1,135 @@
+#include <stdio.h>
+#include <stdbool.h>
+
+typedef enum
+{
+	OK = 0,
+	Numbers = -1,
+	Margin = -2,
+	Scanning = -3,
+	FileIsEmpty = -4
+} StatesOfStructureErrors;
+
+// все что угодно | число (мб начинается с нуля) | число (мб начинается с нуля)
+
+int main()
+{
+	FILE *file = fopen("order.txt", "r");
+	if (file == NULL)
+	{
+		printf("File not found!\n");
+		return -2;
+	}
+
+	int amountOfProduct = 0;
+	int priceOfProduct = 0;
+	char currentChar = 0;
+	char resultOfScanning = -1;
+	char stateOfStructure = OK; // проверка на правильную структуру файла
```

Comment:
```suggestion
	StatesOfStructureErrors stateOfStructure = OK; // проверка на правильную структуру файла
```
Тип этой переменной должен соответствовать типу значений, которые мы присваиваем.